ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/facade/DefaultAuditor.java

```diff
@@ -48,10 +58,38 @@ public void setup()
     @Override
     public void audit(AuditEntry logEntry)
     {
-        if (!filter.isFiltered(logEntry))
+        if (shouldAudit(logEntry))
+        {
+            performAudit(logEntry);
+        }
+    }
+
+    private boolean shouldAudit(AuditEntry logEntry)
+    {
+        long start = System.nanoTime();
+        try
+        {
+            return !filter.isFiltered(logEntry);
+        }
+        finally
+        {
+            long end = System.nanoTime();
+            auditMetrics.filterAuditRequest(end - start, TimeUnit.NANOSECONDS);
+        }
+    }
+
+    private void performAudit(AuditEntry logEntry)
+    {
+        long start = System.nanoTime();
+        try
         {
             AuditEntry obfuscatedEntry = obfuscator.obfuscate(logEntry);
             logger.log(obfuscatedEntry);
         }
+        finally
```

Comment:
My intention with this was to have a way to see if and where we spend time when audit logging without having to run flight recordings. I think the logging framework might throw exceptions, i.e. when disk space is full, but I'm not sure.

If we are using some logging framework that could timeout and throws an exception because of it, then it might be interesting to see those tail latencies. But at that point we would probably see an exception somewhere.

I'll keep it as is for now unless you have a strong opinion on it.